clang/runtime/dpct-rt/include/dpct/bindless_images.hpp

```diff
@@ -146,6 +146,215 @@ class image_mem_wrapper {
+  /// @return size
+  unsigned long long get_res_size() { return size_in_bytes; }
+
+  /// @brief flags getter
+  /// @return flags
+  int get_flags() { return flags; }
+
+#ifdef _WIN32
+  /// @brief win32 handle external_mem_descriptor getter
+  /// @return external_mem_descriptor made from win32 handle resource
+  auto get_ext_res_win32_handle_desc() {
+    return sycl::ext::oneapi::experimental::external_mem_descriptor<
+        sycl::ext::oneapi::experimental::resource_win32_handle>{
+        {get_win32_handle()}, get_handle_type(), get_res_size()};
+  }
+
+  /// @brief win32 obj name external_mem_descriptor getter
+  /// @return external_mem_descriptor made from win32 obj name resource
+  auto get_ext_res_win32_name_desc() {
+    return sycl::ext::oneapi::experimental::external_mem_descriptor<
+        sycl::ext::oneapi::experimental::resource_win32_name>{
+        {get_win32_obj_name()}, get_handle_type(), get_res_size()};
+  }
+#else
+  /// @brief fd external_mem_descriptor getter
+  /// @return external_mem_descriptor made from fd resource
+  auto get_ext_res_fd_desc() {
+    return sycl::ext::oneapi::experimental::external_mem_descriptor<
+        sycl::ext::oneapi::experimental::resource_fd>{
+        {get_fd_handle()}, get_handle_type(), get_res_size()};
+  }
+#endif // _WIN32
+};
+
+/// @brief  The wrapper structure for external image memory desc
+class external_mem_img_desc {
+  sycl::range<3> size;
+  image_channel channel;
+  unsigned int num_levels;
+  sycl::ext::oneapi::experimental::image_type type;
+
+public:
+  external_mem_img_desc() {
+    size = sycl::range<3>{0, 0, 0};
+    channel = image_channel::create<float>();
+    num_levels = 1;
+    type = sycl::ext::oneapi::experimental::image_type::standard;
+  }
+
+  /// @brief size setter
+  /// @param img_dims image dimensions
+  void set_size(sycl::range<3> img_dims) { size = img_dims; }
+
+  /// @brief image_channel setter
+  /// @param img_ch image channel values
+  void set_image_channel(image_channel img_ch) { channel = img_ch; }
+
+  /// @brief num_levels setter
+  /// @param numLevels number of levels in mimap image
+  void set_num_levels(unsigned int numLevels) { num_levels = numLevels; }
+
+  /// @brief type setter
+  /// @param img_type type of the image
+  void set_image_type(sycl::ext::oneapi::experimental::image_type img_type) {
+    type = img_type;
+  }
+
+  /// @brief size getter
+  /// @return size
+  sycl::range<3> get_size() { return size; }
+
+  /// @brief image_channel getter
+  /// @return image_channel
+  image_channel get_image_channel() { return channel; }
+
+  /// @brief num_levels getter
+  /// @return num_levels
+  unsigned int get_num_levels() { return num_levels; }
+
+  /// @brief type getter
+  /// @return type
+  sycl::ext::oneapi::experimental::image_type get_image_type() { return type; }
+
+  /// @brief generate sycl image_descriptor
+  /// @return image_descriptor
+  sycl::ext::oneapi::experimental::image_descriptor get_sycl_obj() {
```

Comment:
this helper function is not necessary.

Reply:
This function helps simplify the SYCL replacement of `cudaExternalMemoryGetMappedMipmappedArray()` below:

`cudaExternalMemoryGetMappedMipmappedArray(&mipmap, extMem, &mipmappedArrDesc);`

to

`mipmap = new dpct::experimental::image_mem_wrapper(`
`                       (&mipmappedArrDesc)->get_sycl_obj(),`
`                       sycl::ext::oneapi::experimental::map_external_image_memory(`
`                           extMem,`
`                           (&mipmappedArrDesc)->get_sycl_obj(),`
`                           q_ct1));`

Reply:
There is cost to maintain helper function,  prefer to use sycl API directly.